sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py

```diff
@@ -579,40 +563,46 @@ def move_to_joint_value_target_unsafe(self, joint_states, time=0.002,
         """
         # self._update_default_trajectory()
         # self._set_targets_to_default_trajectory(joint_states)
+        goals = {}
         joint_states_cpy = copy.deepcopy(joint_states)
 
         if angle_degrees:
             joint_states_cpy.update((joint, radians(i))
                                     for joint, i in joint_states_cpy.items())
 
-        goal = FollowJointTrajectoryGoal()
-        goal.trajectory.joint_names = list(joint_states_cpy.keys())
-        point = JointTrajectoryPoint()
-        point.time_from_start = rospy.Duration.from_sec(time)
-        for key in goal.trajectory.joint_names:
-            point.positions.append(joint_states_cpy[key])
+        for i, value in enumerate(self._controllers.values()):
+            goal = FollowJointTrajectoryGoal()
+            goal.trajectory.joint_names = list(value)
+            point = JointTrajectoryPoint()
+            point.time_from_start = rospy.Duration.from_sec(time)
+            for key in goal.trajectory.joint_names:
+                point.positions.append(joint_states_cpy[key])
 
-        goal.trajectory.points = []
-        goal.trajectory.points.append(point)
+            goal.trajectory.points = []
+            goal.trajectory.points.append(point)
+            goals.update({i:goal})
 
-        self._call_action(goal)
+        self._call_action(goals)
 
         if not wait:
             return
 
-        if not self._client.wait_for_result():
-            rospy.loginfo("Trajectory not completed")
+        for i in self._clients.keys():
+            if not self._clients[i].wait_for_result():
+                rospy.loginfo("Trajectory not completed")
 
     def action_is_running(self):
         return self._action_running
 
     def _action_done_cb(self, terminal_state, result):
         self._action_running = False
 
-    def _call_action(self, goal):
-        self._set_up_action_client()
+    def _call_action(self, goals):
+        self._set_up_action_client(self._controllers)
         self._action_running = True
-        self._client.send_goal(goal, self._action_done_cb)
+
+        for i, client in enumerate(self._clients.keys()):
+            self._clients[client].send_goal(goals[i], self._action_done_cb)
```

Comment:
Using the same callback function for both goals can be an issue. When the first goal finishes we will have `self._action_running = True` which will be misleading if the second (or nth) action hasn't finished.